portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,29 +14,84 @@
 
 package com.google.sps.servlets;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.sps.data.Comment;
+import com.google.gson.Gson;
 import java.io.IOException;
+import java.util.Arrays;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
-import java.util.Arrays;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+
+
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-    private List<String> names;
+  /** The max amount of comments that will be displayed. */
+  private int maxComments = 5;
+
+  /** 
+   * Convert List object into JSON.
+   * @param {!List} jsonToParse List object to convert into JSON.
+   */
+  private final String convertToJson(List jsonToParse) {
+    Gson gson = new Gson();
+    String json = gson.toJson(jsonToParse);
+    return json;
+}
+
+  /** Read the data from the datastore and write it into /data as json. */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").addSort("timeStamp", SortDirection.DESCENDING);
 
-    @Override
-    public void init() {
-        names = new ArrayList<String>(Arrays.asList("Erin", "Ruchi", "Owen"));
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    List<Entity> resultsList = results.asList(FetchOptions.Builder.withLimit(maxComments));
+    
 
-    @Override
-    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-        String name = names.get((int) (Math.random() * names.size()));
-        response.setContentType("text/html;");
-        response.getWriter().println("Hello " + name + "!");
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : resultsList) {
+            // Entity entity = results.next();
+            long id = entity.getKey().getId();
+            String author = (String) entity.getProperty("author");
+            String comment = (String) entity.getProperty("comment");
+            String timestamp = (String) entity.getProperty("timeStamp");
+
+            Comment commentObject = new Comment(id, author, comment, timestamp);
+            comments.add(commentObject);
```

Comment:
looks like there's too much indent. 2 spaces

Reply:
ah darn, switching from js to java is hard on the brain xC